api/desecapi/views.py

```diff
@@ -349,7 +345,8 @@ def _find_ip_v6(self, request):
         return self.find_ip(request, ['myipv6', 'ipv6', 'myip', 'ip'], version=6)
 
     def get(self, request, *_):
-        domain = self._find_domain(request)
+        qname, domain = self._find_domain(request)
+        subname = qname.rpartition(domain.name)[0]
```

Comment:
maybe move this logic into `_find_domain`, that way it seems to me that `_find_domain` has a nice self-contained interface?

Reply:
refactored